src/dataviews/dataview-model-base.js

```diff
@@ -76,6 +78,7 @@ module.exports = Model.extend({
 
   _initBinds: function () {
     this.listenTo(this.layer, 'change:visible', this._onLayerVisibilityChanged);
+    this.listenTo(this.layer, 'change:source', this._onLayerVisibilityChanged);
```

Comment:
Aren't we taking care of that [here](https://github.com/CartoDB/cartodb.js/pull/1359/files/ab4d43799e3ba602de36efbe15339ecbf76f0f2f#diff-805364e485d740e26e33839755aab753R97)?